### PR TITLE
Provide errors in all cases that local functions name conflict with other names

### LINKED ISSUE #13913: Remove BoundVoid
The need for the bound node `BoundVoid` is an artifact of the fact that deconstruction is prematurely lowered. Once it is not prematurely lowered, we can and should remove `BoundVoid`.

### PR DESCRIPTION
**Pull request by agocke**
Also rephrases some existing error messages to incorporate local functions into
the description.

Fixes #13913.

### CODE REVIEW

**Comment by agocke** [comment]
test windows_debug_unit64_prtest please

File: src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs [hunk 1]
```diff
@@ -285,7 +287,9 @@ internal static bool ReportConflictWithParameter(Symbol parameter, Symbol newSym
 
             if (parameterKind == SymbolKind.TypeParameter)
             {
-                if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local)
+                if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local ||
+                    (newSymbolKind == SymbolKind.Method && 
+                     ((MethodSymbol)newSymbol).MethodKind == MethodKind.LocalFunction))
                 {
                     // CS0412: 'X': a parameter or local variable cannot have the same name as a method type parameter
                     diagnostics.Add(ErrorCode.ERR_LocalSameNameAsTypeParam, newLocation, name);
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs]: Consider updating the comment in the use of `ERR_LocalDuplicate` in `LocalScopeBinder` as well.

**Comment by cston** [comment]
LGTM

File: src/Compilers/Core/Portable/project.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿{
   "supports": {},
   "dependencies": {
+    "Microsoft.NETCore.Portable.Compatibility": "1.0.1",
```
Comment (agocke) [inline src/Compilers/Core/Portable/project.json]: Yeah -- it makes the IDE understand resources and stuff in portable projects.

**Comment by agocke** [comment]
@dotnet/roslyn-compiler One more review please

**Comment by gafter** [comment]
I think you closed the wrong issue number

**Comment by agocke** [comment]
Yup, meant to close #13193, not #13913!

**Pull request merged**